ex1/functions.py

```diff
@@ -90,26 +192,127 @@ def f():
+def args_to_dict(a, b, c):
+    """
+    takes three arguments and returns a dictionary with the position of the
+    arguments as the key (starting at 0) and the argument as the value
+    """
+    return {0:a, 1:b, 2:c}
 
 # BONUS: Write a function named args_to_dict_general that does the same for any
 # number of arguments
+def args_to_dict_general(*args):
+    """
+    takes n arguments and returns a dictionary with the position of the
+    arguments as the key (starting at 0) and the argument as the value
+    """
+    i = 0
+    abc = {}
+    for arg in args:
+        abc[i] = arg
+        i += 1
+    return abc
 
 # Define a function named lists_to_dict that takes two lists of equal lenght
 # named keys and values and builds a dictionary out of them.
+def lists_to_dict(list1, list2):
+    """
+    builds a dictionary of two lists of equal length
+    Parameters
+    ----------
+    list1: list
+        first content (keys)
+    list2: list
+        second content (values)
+    Returns
+    ----------
+    abc: dict
+        of the two lists
+    """
+    i = 0   
+    abc = {}
+    for vals in list1:
+        abc[vals] = list2[i]
+        i += 1
+    return abc
 
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
+def search_list(lista, listb):
+    """
+    creates a dictionary of elements of listb found in lista with the index where
+    found as key (if nothing found the dictionary is empty)
+    Parameters
+    ----------
+    lista: list
+        base content
+    listb: list
+        searched content
+    Returns
+    ----------
+    abc: dict
+        of the content found in both lists
+    """
+    abc = {}    
+    for valb in listb:
+        if valb in lista:
+            abc[lista.index(valb)] = valb
+    return abc
 
 # Define a function named dict_to_string that takes a dictionary and a
 # separator string. The function should only take elements out of the
 # dictionary whose value is a string and then return a single string containing
 # the strings stored in the dictionary seperated by the separator string.
 # Return an empty string if there are no strings in the dictionary.
+def dict_to_string(dic, sep):
+    """
+    creates a string of all strings found in the dictionary separated by the given
+    string (if nothing found the string is empty)
+    Parameters
+    ----------
+    dic: dict
+        base content
+    sep: str
+        seperator string
+    Returns
+    ----------
+    answer: string
+        sum of all strings from the dic, seperated with the sep string
+    """
+    answer = ""
+    for key,value in dic.items():
+        if type(value) is str:
+            answer = answer + value + sep
+    if answer[-1:] == sep:
```

Comment:
S.g. Herr Paulik,
da ich in den letzten beiden Einheiten nicht anwesend sein konnte, möchte ich mich erkundigen, ob die VU mit der Abgabe der beiden Aufgaben erledigt ist, oder ob diese noch an Hand Ihrer Kommentare verbessert werden müssen bzw. ob es noch eine dritte Aufgabe gibt. Vielen Dank.

MfG,
Georg Engelhardt

----- Ursprüngliche Nachricht -----
Von: "Christoph Paulik" <notifications@github.com>
Gesendet: ‎01.‎02.‎2016 15:32
An: "TUW-GEO-python-intro/Exercise-1" <Exercise-1@noreply.github.com>
Cc: "e1127200" <e1127200@student.tuwien.ac.at>
Betreff: Re: [Exercise-1] add all four files (#8)

In ex1/functions.py:
> +    Parameters
> +    ----------
> +    dic: dict
> +        base content
> +    sep: str
> +        seperator string
> +    Returns
> +    ----------
> +    answer: string
> +        sum of all strings from the dic, seperated with the sep string
> +    """
> +    answer = ""
> +    for key,value in dic.items():
> +        if type(value) is str:
> +            answer = answer + value + sep
> +    if answer[-1:] == sep:
If you build a list and use the str.join method you can avoid checks like this. See #20
—
Reply to this email directly or view it on GitHub.